main/src/ca/uwaterloo/flix/api/lsp/provider/SymbolProvider.scala

```diff
@@ -27,31 +27,77 @@ object SymbolProvider {
       return Nil
     }
 
-    root.enums.filter {
-      case (_, enum) => enum.loc.source.name == uri
-    }.map {
-      case (_, enum) => enumToDocumentSymbol(enum)
-    }.toList
+    val enums = root.enums.values.collect { case enum if enum.loc.source.name == uri => mkEnumDocumentSymbol(enum) }
+    val defs = root.defs.values.collect { case d if d.sym.loc.source.name == uri => mkFunctionDocumentSymbol(d) }
+    val classes = root.classes.values.collect { case c if c.sym.loc.source.name == uri => mkClassDocumentSymbol(c) }
+    (classes ++ defs ++ enums).toList
+  }
+
+  /**
+    * Returns an Interface DocumentSymbol from a Class node.
+    * It navigates the AST and adds Sig and TypeParam of c and as children DocumentSymbols.
+    */
+  private def mkClassDocumentSymbol(c: TypedAst.Class): DocumentSymbol = c match {
+    case TypedAst.Class(doc, _, sym, tparam, _, signatures, _, loc) => DocumentSymbol(
+      sym.name,
+      Some(doc.lines.mkString(" ")),
+      SymbolKind.Interface,
+      Range.from(loc),
+      Range.from(loc),
+      Nil,
+      signatures.map(mkSignatureDocumentSymbol) :+ mkTypeParamDocumentSymbol(tparam),
+    )
+  }
+
+  /**
+    * Returns a TypeParameter DocumentSymbol from a TypeParam node.
+    */
+  private def mkTypeParamDocumentSymbol(t: TypedAst.TypeParam) = t match {
+    case TypedAst.TypeParam(name, tpe, loc) => DocumentSymbol(
+      name.name, Some(tpe.kind.toString ()), SymbolKind.TypeParameter, Range.from(loc), Range.from(loc), Nil, Nil,
+    )
+  }
+
+  /**
+    * Returns a Method DocumentSymbol from a Sig node.
+    */
+  private def mkSignatureDocumentSymbol(s: TypedAst.Sig): DocumentSymbol = s match {
+    case TypedAst.Sig(sym, spec, _) => DocumentSymbol(
+      sym.name, Some(spec.doc.lines.mkString(" ")), SymbolKind.Method, Range.from(sym.loc), Range.from(sym.loc), Nil, Nil,
```

Comment:
Here and in a few other places, instead of `spec.doc.lines.mkString(...)`, you should use `doc.text`.

Reply:
Thanks!